pkg/graveler/hooks_handler.go

```diff
@@ -10,29 +10,59 @@ import (
 type EventType string
 
 const (
-	EventTypePreCommit  EventType = "pre-commit"
-	EventTypePostCommit EventType = "post-commit"
-	EventTypePreMerge   EventType = "pre-merge"
-	EventTypePostMerge  EventType = "post-merge"
+	EventTypePreCommit        EventType = "pre-commit"
+	EventTypePostCommit       EventType = "post-commit"
+	EventTypePreMerge         EventType = "pre-merge"
+	EventTypePostMerge        EventType = "post-merge"
+	EventTypePreCreateTag     EventType = "pre-create-tag"
+	EventTypePostCreateTag    EventType = "post-create-tag"
+	EventTypePreDeleteTag     EventType = "pre-delete-tag"
+	EventTypePostDeleteTag    EventType = "post-delete-tag"
+	EventTypePreCreateBranch  EventType = "pre-create-branch"
+	EventTypePostCreateBranch EventType = "post-create-branch"
+	EventTypePreDeleteBranch  EventType = "pre-delete-branch"
+	EventTypePostDeleteBranch EventType = "post-delete-branch"
 )
 
+// HookRecord is an aggregation of all necessary fields for all event types
+// Required fields for all event types:
+// RunID
+// EventType
+// RepositoryID
+// StorageNamespace
+// SourceRef - The reference from which the actions files are read from
+// Event specific fields:
+// BranchID - Relevant for all event types except tags. For merge events this will be the ID of the destination branch
+// Commit - Relevant only for commit and merge events. In both it will contain the new commit data created from the operation
+// CommitID - Not relevant in delete branch. In commit and merge will not exist in pre-action. In post actions will contain the new commit ID
+// PreRunID - Exists only in post actions. Contains the ID of the pre-action associated with this post-action
+// TagID - Exists only in tag actions.
```

Comment:
Thanks for the documentation!
Usually field comments go above the actual field.

Reply:
In this case - do you think it's better to leave it like this or to break the documentation per field?

Reply:
I think it's better to have per field.